contracts/dependency-registry/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1127 @@
+     * dependency `_dependencyTypes` at script index `_index`.
+     */
+    function getDependencyScriptBytecodeAddressAtIndex(
+        bytes32 _dependencyType,
+        uint256 _index
+    ) external view returns (address) {
+        return
+            dependencyDetails[_dependencyType].scriptBytecodeAddresses[_index];
+    }
+
+    /**
+     * @notice Returns script for dependency `_dependencyType` at script index `_index`.
+     * @param _dependencyType dependency to be queried.
+     * @param _index Index of script to be queried.
+     */
+    function getDependencyScriptAtIndex(
+        bytes32 _dependencyType,
+        uint256 _index
+    ) external view returns (string memory) {
+        Dependency storage dependency = dependencyDetails[_dependencyType];
+        // If trying to access an out-of-index script, return the empty string.
+        if (_index >= dependency.scriptCount) {
+            return "";
+        }
+
+        return dependency.scriptBytecodeAddresses[_index].readFromBytecode();
+    }
+
+    /**
+     * @notice Returns the dependency type for a given project (`projectId`)
+     * on a given core contract (`_contractAddress`). If no override is set,
+     * the core contract is called to retrieve the script type and version as
+     * dependency type. For any contract earlier than v3, that does not have
+     * an override set, this will revert.
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to return dependency type for.
+     * @return dependencyType Dependency type used by project.
+     */
+    function getDependencyTypeForProject(
+        address _contractAddress,
+        uint256 _projectId
+    ) external view returns (string memory) {
+        _onlySupportedCoreContract(_contractAddress);
+
+        bytes32 dependencyType = projectDependencyTypeOverrides[
+            _contractAddress
+        ][_projectId];
+        if (dependencyType != bytes32(0)) {
+            return dependencyType.toString();
+        }
+
+        try
+            IDependencyRegistryCompatibleV0(_contractAddress)
+                .projectScriptDetails(_projectId)
+        returns (
+            string memory scriptTypeAndVersion,
+            string memory,
+            uint256 scriptCount
+        ) {
+            return scriptTypeAndVersion;
+        } catch {
+            revert(
+                "Contract does not implement projectScriptDetails and has no override set."
+            );
+        }
+    }
+
+    /**
+     * @notice Converts string to bytes32.
+     * https://ethereum.stackexchange.com/a/9152
+     * @param source string to be converted.
+     * @return result bytes32 representation of string.
+     */
+    function stringToBytes32(
+        string memory source
+    ) public pure returns (bytes32 result) {
+        bytes memory tempEmptyStringTest = bytes(source);
+        if (tempEmptyStringTest.length == 0) {
+            return 0x0;
+        }
+
```

Comment:
This should perform the following check to ensure the string is not too long to fit into a bytes32:
```
require(bytes(source).length > 31, "String too long");
```